### DISCUSSION TITLE
IPAM status

### DISCUSSION THREAD

**Discussion by jk-onnet** [netbox-community/netbox#9178] [Q&A]
Can I ask how you all define an IP address status?

Server IP address 
1. configured + reachable == active (OK)
2. configured + unreachable == active (?) reserved (?)
3. interface disconnected + unreachable == deprecated (?) reserved (?)
4. server disconnected + unreachable == deprecated (?) active (?)

I want to audit them and update the NetBox value based on current NetBox value using configuration management tools. I'm not sure in general how people define the state.

**Comment by candlerb** [comment]
I don't do any of that at the IP address level.

If the IP address is assigned to a device or VM, then it's "Active".  If it's down or disconnected I would simply mark the Device or VM as "Offline" or "Failed", not the IP address. The IP address is still "Active" because it's actively assigned to some entity, and therefore is not available for assignment to anything else - even if that entity is not connected right now.

If you wanted to record the presence or absence of an IP address based on active network testing, then you could use custom fields for that.  But do bear in mind that Netbox is really intended as a source of truth for network provisioning, not as a portal onto live network status.  See https://netbox.readthedocs.io/en/stable/#serve-as-a-source-of-truth

I don't use my monitoring tools to update Netbox.  Rather, I use Netbox to configure the monitoring tools (e.g. setting a device to "Offline" or "Failed" in Netbox removes it from monitoring)
